### DISCUSSION TITLE
GH project's settings

### DISCUSSION THREAD

**Discussion by fardavide** [androiddevelopersitalia/aditogether#60] [General]
I would like to use this discussion to decide about the GitHub project's settings.

@mcatta @fondesa

**Comment by fardavide** [comment]
**Auto-delete branches on PR merge**
I believe it's good practice to keep the repository clean from unnecessary branches, so I would like to have the auto-delete enabled by default

**Reply by fondesa** [reply]
+1 on this, if it's really needed, the branch can be restored anyway (never had to do that honestly since I use auto-delete on my own as well)

**Reply by mcatta** [reply]
+1

**Comment by fardavide** [comment]
**Update branches by rebase**
Currently, updating branches creates a merge commit `main -> branch`.
I think it's good practice to keep the branches history linear, so I would enable rebase by default and forbid merges

**Reply by fondesa** [reply]
Do you mean updating the branch using the GH UI?

**Reply by fardavide** [reply]
Exactly

**Reply by fondesa** [reply]
Okok, +1 so

**Comment by giuseppesorce** [comment]
i use Gtihub for my personal project and bitbucket for company projects so I don't know how does githuib work with PR but I agree auto-delete branch in generally.
